lib/dfdutils/queries.c

```diff
@@ -19,6 +19,49 @@
 #include <KHR/khr_df.h>
 #include "dfd.h"
 
+#if !defined(BITFIELD_ORDER_FROM_MSB)
+// Most compilers, including all those tested so far, including clang, gcc
+// and msvc, order bitfields from the lsb so these struct declarations work.
+// Could this be because I've only tested on little-endian machines?
+// These allow the debugger to print the DFD contents in a human meaningful
+// way making debugging much easier.
+struct sampleType {
+    uint32_t bitOffset: 16;
+    uint32_t bitLength: 8;
+    uint32_t channelType: 8; // Includes qualifiers
+    uint32_t samplePosition0: 8;
+    uint32_t samplePosition1: 8;
+    uint32_t samplePosition2: 8;
+    uint32_t samplePosition3: 8;
+    uint32_t lower;
+    uint32_t upper;
+};
+
+struct BDFD {
+    uint32_t vendorId: 17;
+    uint32_t descriptorType: 15;
+    uint32_t versionNumber: 16;
+    uint32_t descriptorBlockSize: 16;
+    uint32_t model: 8;
+    uint32_t primaries: 8;
+    uint32_t transfer: 8;
+    uint32_t flags: 8;
+    uint32_t texelBlockDimension0: 8;
+    uint32_t texelBlockDimension1: 8;
+    uint32_t texelBlockDimension2: 8;
+    uint32_t texelBlockDimension3: 8;
+    uint32_t bytesPlane0: 8;
+    uint32_t bytesPlane1: 8;
+    uint32_t bytesPlane2: 8;
+    uint32_t bytesPlane3: 8;
+    uint32_t bytesPlane4: 8;
+    uint32_t bytesPlane5: 8;
+    uint32_t bytesPlane6: 8;
+    uint32_t bytesPlane7: 8;
+    struct sampleType samples[6];
+};
+#endif
+
```

Comment:
I'm concerned about introducing yet another definition for BDFD types. We already have too many, and this is prone to be the source of errors. Can't the unit test simply use the one in e.g. imageio?

Regarding big-endian support, we did find various places in libktx where we had concerns whether it actually works on big-endian machines, so I think big-endian support is a wider open question, but the definitions here wouldn't work on big-endian.

Reply:
As the comment says, its just to help with debugging. It isn't used. This file is actually in the dfdutils so it is difficult to share any header except a public header. Possibly I could add this struct definition to `dfd.h`. What do you think?

Reply:
Well, if this is only expected to be used by tests, I'd rather put it there. Putting it into dfdutils makes it look like an official definition to be used by dfdutils library users.

Reply:
I'll delete it. It's at the top of `texture2.c` if ever needed again.